### PR TITLE
DOC: Document DataFrame.categorize

### PR DESCRIPTION
**Pull request by jseabold**

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -1527,6 +1527,27 @@ def groupby(self, key, **kwargs):
...
+        columns : list, optional
+            A list of column names to convert to the category type. By
+            default any column with an object dtype is converted to a
+            categorical.
+        kwargs
+            Keyword arguments are passed on to compute.
+
+        Notes
+        -----
+        Whenever possible, dealing with categorical type variabless, is much
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Perhaps mention when categoricals are nicer rather than "whenever possible"

**Comment by mrocklin** [comment]
Hrm, interesting test failure here. Almost certainly not your fault.  Pinging @jcrist

**Comment by jcrist** [comment]
That's an odd error, looks like something (worker or scheduler) tried to push to a closed pipe in the queue. This can happen if an error is thrown in the receiving process (http://stackoverflow.com/questions/23482115/eoferror-with-multiprocessing-manager). I'm unable to reproduce locally, but restarted the build.

**Comment by mrocklin** [comment]
Fixed typo, squashed, and merged in https://github.com/dask/dask/commit/a90a6e3b605c5c803cfc7fb5092191b037428908

**Pull request closed without merging**